src/protocolgame.cpp

```diff
@@ -2882,6 +2921,25 @@ void ProtocolGame::sendHouseWindow(uint32_t windowTextId, const std::string& tex
 	writeToOutputBuffer(msg);
 }
 
+void ProtocolGame::sendCombatAnalyzer(CombatType_t type, int32_t amount, DamageAnalyzerImpactType impactType, const std::string &target)
+{
+	NetworkMessage msg;
+	msg.addByte(0xCC);
+	msg.addByte(impactType);
+	msg.add<uint32_t>(amount);
+
+	switch (impactType)	{
+		case RECEIVED:
+			msg.addByte(getClientElement(type));
+			msg.addString(target);
+			break;
+		case DEALT:
+			msg.addByte(getClientElement(type));
+			break;
```

Comment:
Don't forget to implement `default: break;`

Reply:
Not essential but wouldn't hurt. I suggest adding newlines between cases though